cmd/schemagen/schemagen.go

```diff
@@ -0,0 +1,155 @@
+package main
+
+import (
+	"bytes"
+	"embed"
+	"flag"
+	"fmt"
+	"github.com/scylladb/gocqlx/v2"
+	"go/format"
+	"html/template"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"unicode"
+
+	"github.com/gocql/gocql"
+	_ "github.com/scylladb/gocqlx/v2/table"
+)
+
+var (
+	cmd          = flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	flagCluster  = cmd.String("cluster", "127.0.0.1", "a comma-separated list of host:port tuples")
+	flagKeyspace = cmd.String("keyspace", "system", "keyspace to inspect")
+	flagPkgname  = cmd.String("pkgname", "models", "the name you wish to assign to your generated package")
+	flagOutput   = cmd.String("output", "models", "the name of the folder to output to")
+)
+
+var (
+	//go:embed keyspace.tmpl
+	rootFs embed.FS
+)
+
+func main() {
+	assert(cmd.Parse(os.Args[1:]), "can't parse flags")
+
+	schemagen()
+}
+
+func schemagen() {
+	writeToFile(
+		renderTemplate(
+			fetchMetadata(
+				createSession())))
+}
+
+func assert(err error, msg string) {
+	if err != nil {
+		fmt.Println(msg)
+		panic(err)
+	}
+}
+
+func writeToFile(b []byte) {
+	outputPath := path.Join(*flagOutput, *flagPkgname+".go")
+	ensureDir(outputPath)
+
+	f, err := os.OpenFile(outputPath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0755)
+	assert(err, "unable to open output file")
+	defer f.Close()
+
+	_, err = f.Write(b)
+	assert(err, "unable to write to output file")
+
+	fmt.Println("File written to", outputPath)
+}
+
+func fetchMetadata(s *gocqlx.Session) *gocql.KeyspaceMetadata {
+	md, err := s.KeyspaceMetadata(*flagKeyspace)
+	assert(err, "unable to fetch keyspace metadata")
+
+	return md
+}
+
+func renderTemplate(md *gocql.KeyspaceMetadata) []byte {
+	t, err := template.
+		New("keyspace.tmpl").
+		Funcs(template.FuncMap{"camelize": Camelize}).
+		ParseFS(rootFs, "keyspace.tmpl")
+
+	assert(err, "unable to parse models template")
+
+	buf := &bytes.Buffer{}
+	data := map[string]interface{}{
+		"PackageName": *flagPkgname,
+		"Tables":      md.Tables,
+	}
+
+	err = t.Execute(buf, data)
+	assert(err, "unable to execute models template")
+
+	res, err := format.Source(buf.Bytes())
+	assert(err, "template output is not a valid go code")
+
+	return res
+}
+
+func createSession() *gocqlx.Session {
+	cluster := createCluster()
+	s, err := gocqlx.WrapSession(cluster.CreateSession())
+	assert(err, "unable to create scylla session")
+	return &s
+}
+
+func createCluster() *gocql.ClusterConfig {
+	clusterHosts := getClusterHosts()
+	return gocql.NewCluster(clusterHosts...)
+}
+
+func getClusterHosts() []string {
+	return strings.Split(*flagCluster, ",")
+}
+
+func Camelize(s string) string {
```

Comment:
We probably do not want to export it and maybe we want to use a library instead.